src/pages/index.js

```diff
@@ -1,118 +1,174 @@
-  profileFormValidator.resetValidation();
-  profileModal.open();
+    const userData = userInfo.getUserInfo();
+    nameInput.value = userData.name;
+    descriptionInput.value = userData.about;
+    profileModal.open();
 });
 
-addCardButton.addEventListener("click", () => {
-  newCardModal.open();
+const newCardModal = new PopupWithForm("#add-card-modal", (data) => {
+    newCardModal.renderLoading(true);
+    api.createCard({ name: data.title, link: data.url })
+        .then((newCardData) => {
+            const cardElement = createCard(newCardData);
+            section.addItem(cardElement);
+            newCardModal.close();
+        })
+        .catch(err => console.error("Error creating card:", err))
+        .finally(() => newCardModal.renderLoading(false));
 });
 
+newCardModal.setEventListeners();
+addCardButton.addEventListener("click", () => newCardModal.open());
+
+const deleteCardPopup = new PopupWithConfirmation("#confirm-delete-modal");
+deleteCardPopup.setEventListeners();
+
+function handleDeleteCard(cardId, cardElement) {
+    deleteCardPopup.open(() => {
+        api.deleteCard(cardId)
+            .then(() => {
+                cardElement.remove();
+                deleteCardPopup.close();
+            })
+            .catch(err => console.error("Error deleting card:", err));
+    });
+}
 
-const addCardFormValidator = new FormValidator(formValidationSettings, addCardForm);
-const profileFormValidator = new FormValidator(
-  formValidationSettings,
-  editProfileForm
-);
-
-addCardFormValidator.enableValidation();
-profileFormValidator.enableValidation();
-
-
-const imageModal = new PopupWithImage("#modal__preview-card");
-
-imageModal.setEventListeners();
-
+const avatarModal = new PopupWithForm("#change-avatar-modal", (data) => {
+    avatarModal.renderLoading(true);
+    api.changeAvatar(data.url)
+        .then((updatedData) => {
+            userInfo.setAvatar(updatedData.avatar);
+            avatarModal.close();
+        })
+        .catch((err) => console.error("Error updating avatar:", err))
+        .finally(() => avatarModal.renderLoading(false));
+});
 
-const newCardModal = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
-);
+avatarModal.setEventListeners();
 
-newCardModal.setEventListeners();
+document.querySelector(".profile__edit-image-button").addEventListener("click", () => {
+    avatarModal.open();
+});
 
+function handleLikeClick(cardId, isLiked) {
+    if (isLiked) {
+        api.removeLike(cardId)
+            .catch(err => console.error("Error removing like:", err));
+    } else {
+        api.addLike(cardId)
+            .catch(err => console.error("Error adding like:", err));
+    }
+}
 
-const profileModal = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileFormSubmit
-);
+const profileFormValidator = new FormValidator(formValidationSettings, editProfileForm);
+const addCardFormValidator = new FormValidator(formValidationSettings, addCardForm);
```

Comment:
**COULD BE IMPROVED**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```